src/services/shortLink/service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ClientHostManager,
+  Context,
+  PermissionLevel,
+  ShortLinkPatchPayload,
+  ShortLinkPostPayload,
+} from '@graasp/sdk';
+
+import { UnauthorizedMember } from '../../utils/errors';
+import { Repositories } from '../../utils/repositories';
+import ItemService from '../item/service';
+import { Actor } from '../member/entities/member';
+
+export class ShortLinkService {
+  private itemService: ItemService;
+
+  public constructor(itemService: ItemService) {
+    this.itemService = itemService;
+  }
+
+  async post(member: Actor, repositories: Repositories, shortLink: ShortLinkPostPayload) {
+    const { shortLinkRepository } = repositories;
+
+    if (!member) throw new UnauthorizedMember();
+    // check that the member can admin the item to be allowed to create short link
+    await this.itemService.get(member, repositories, shortLink.item.id, PermissionLevel.Admin);
+
+    return shortLinkRepository.createOne(shortLink);
+  }
+
+  async getOne(repositories: Repositories, alias: string) {
+    const { shortLinkRepository } = repositories;
+    return shortLinkRepository.get(alias);
+  }
+
+  async getAllForItem(member: Actor, repositories: Repositories, itemId: string) {
+    const { shortLinkRepository } = repositories;
+
+    if (!member) throw new UnauthorizedMember();
+    // check that the member can admin the item to be allowed to create short link
+    await this.itemService.get(member, repositories, itemId, PermissionLevel.Admin);
+
+    return shortLinkRepository.getItem(itemId);
+  }
+
+  async getRedirection(repositories: Repositories, alias: string) {
+    const shortLink = await this.getOne(repositories, alias);
+    const clientHostManager = ClientHostManager.getInstance();
+
+    return clientHostManager
+      .getItemLink(shortLink.platform as Context, shortLink.item.id)
```

Comment:
I guess you're receiving a `URL` object. Maybe it's clearer if the function name is `getItemAsURL`. And maybe provide also `getItemLink` that returns a string link?